### PR TITLE
CME credit progress

### PR DESCRIPTION
**Pull request by nburka**

### CODE REVIEW

File: CME/dataobjects/CMEAccount.php [hunk 1]
```diff
@@ -177,6 +178,34 @@ public function getAvailableCMECredits()
...
+		);
+
+		return SwatDB::query(
+			$this->app->db,
+			$sql,
+			SwatDBClassMap::get('CMEAccountCMEProgressWrapper');
+		)->getFirst();
+	}
+
+	// }}}
```
Comment (charleswaddell) [inline CME/dataobjects/CMEAccount.php]: Do we want to make this method so you can pass in multiple credits?

Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: We discussed this in person. The answer is "no". We want to verify if credits belong to the same progress, or if no progress has started.

File: CME/dataobjects/CMEAccount.php [hunk 2]
```diff
@@ -177,6 +178,34 @@ public function getAvailableCMECredits()
...
+		$sql = sprintf(
+			'select AccountCMEProgress.*
+			from AccountCMEProgress
+			where AccountCMEProgress.account = %s
+			and AccountCMEProgress.id in (
+				select progress from AccountCMEProgressCreditBinding
+				where AccountCMEProgressCreditBinding.credit = %s
+			)',
+			$this->db->quote($this->id, 'integer'),
+			$this->db->quote($credit->id, 'integer'),
```
Comment (charleswaddell) [inline CME/dataobjects/CMEAccount.php]: Remove comma.

File: CME/dataobjects/CMEAccount.php [hunk 3]
```diff
@@ -177,6 +178,34 @@ public function getAvailableCMECredits()
...
+		);
+
+		return SwatDB::query(
+			$this->app->db,
+			$sql,
+			SwatDBClassMap::get('CMEAccountCMEProgressWrapper');
```
Comment (charleswaddell) [inline CME/dataobjects/CMEAccount.php]: Remove semi-colon.

File: CME/dataobjects/CMEAccount.php [hunk 4]
```diff
@@ -177,6 +178,34 @@ public function getAvailableCMECredits()
...
+			and AccountCMEProgress.id in (
+				select progress from AccountCMEProgressCreditBinding
+				where AccountCMEProgressCreditBinding.credit = %s
+			)',
+			$this->db->quote($this->id, 'integer'),
+			$this->db->quote($credit->id, 'integer'),
+		);
+
+		return SwatDB::query(
+			$this->app->db,
```
Comment (charleswaddell) [inline CME/dataobjects/CMEAccount.php]: Remove app reference.

File: sql/tables/AccountCMEProgress.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+create table AccountCMEProgress (
+	id serial,
+
+	account integer not null references Account(id) on delete cascade,
+	quiz integer not null references Inquisition(id) on delete cascade,
+	evaluation integer not null references Inquisition(id) on delete cascade,
```
Comment (charleswaddell) [inline sql/tables/AccountCMEProgress.sql]: Both the quiz and evaluation column should allow `null` references.

**Pull request merged**